media-proxy/src/rdma_session.c

```diff
@@ -345,6 +345,11 @@ static void *rx_rdma_frame_thread(void *arg)
     libfabric_ctx *rdma_ctx = s_ctx->rdma_ctx;
     ep_ctx_t *cp_ctx = s_ctx->ep_ctx;
 
+    while (!atomic_load(&s_ctx->shm_ready) && !s_ctx->stop) {
+        INFO("%s memif not ready\n", __func__);
+        usleep(1000);
+    }
+
     printf("%s(%d), start\n", __func__, s_ctx->idx);
     while (!s_ctx->stop) {
```

Comment:
```suggestion
    while (atomic_load_explicit(&s_ctx->shm_ready, memory_order_relaxed) && !s_ctx->stop) {
```

Reply:
looks better but it can take some time between this check and calling rx_rdma_consume_frame() as ep_recv_buf() publishes buffer to receive incoming data into it and waits till transmission is complete. 
So, potentially we can call memif_buffer_alloc() on non existing connection which probably will just return error and nothing tragic would happen, but still, I would prefer to have a check closer to the call to memif_buffer_alloc()